### PR TITLE
Move model_config to common

### PR DESCRIPTION
**Pull request by kpedro88**
Addresses https://github.com/triton-inference-server/server/issues/2759: many of the functions in `model_config.h` are generically useful even on the client side and only depend on the generated protobuf headers. Therefore, it is useful to have them more broadly accessible via the common repo.

A few functions that have non-trivial dependence on the server API or on `constants.h` are kept in the core repo, moved to `model_config_utils`. One definition from `constants.h` is moved to the common repo.

The associated changes in the `model_config.h` header location and the namespace are propagated throughout the core repo.

I checked that these changes compiled successfully by following https://github.com/triton-inference-server/server/blob/main/docs/build.md to get the Docker container and install all needed system dependencies, then built the core repo following https://github.com/triton-inference-server/core#build-the-triton-core-library.

This PR goes with https://github.com/triton-inference-server/common/pull/48.

### CODE REVIEW

File: src/test/CMakeLists.txt [hunk 1]
```diff
@@ -414,6 +412,7 @@ target_link_libraries(
   repo_agent_test
   PRIVATE
     triton-common-error       # from repo-common
+    triton-common-model-config# from repo-common
```
Comment (deadeyegoodwin) [inline src/test/CMakeLists.txt:415]: Add a space before '#'

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/test/CMakeLists.txt (hunk 1 above)
Comment (deadeyegoodwin) [reply src/test/CMakeLists.txt:415]: @jbkyang-nvi  other that this one small change it looks ok to me. But you should review and run CI...

Comment (kpedro88) [reply src/test/CMakeLists.txt]: Done

**Comment by kpedro88** [comment]
superseded by #71

**Pull request closed without merging**